dgraphHelper.js

```diff
@@ -0,0 +1,370 @@
+  MISSING_TYPES.forEach(type => {
+    const DELETED_OBJECT = current_schema.types.find(object => object.name === type);
+    CONFLICTS.push({
+      message: 'This object was deleted.',
+      category: 'types',
+      object: { ...DELETED_OBJECT },
+    });
+  });
+
+  return [CONFLICTS, ADDED];
+}
+
+const diff_schema_checker = (new_schema, current_schema) => {
+  const CONFLICTS = [];
+  const ADDED = [];
+  const PREDICATES_TO_CHECK = [];
+  const MISSING_PREDICATES = [];
+
+  new_schema.schema.forEach(element => {
+    PREDICATES_TO_CHECK.push(element.predicate);
+  });
+
+  current_schema.schema.forEach(element => {
+    if (!PREDICATES_TO_CHECK.includes(element.predicate)) {
+      MISSING_PREDICATES.push(element.predicate);
+    }
+  });
+
+  PREDICATES_TO_CHECK.forEach(predicate => {
+    const NEW_OBJECT = new_schema.schema.find(object => object.predicate === predicate);
+    const CURRENT_OBJECT = current_schema.schema.find(object => object.predicate === predicate);
+    const DIFFERENCES = diff(CURRENT_OBJECT, NEW_OBJECT); // Can have multiple diff for 1 object
+    if (typeof DIFFERENCES !== 'undefined') {
+      DIFFERENCES.forEach(difference => {
+        if (difference.kind === 'E') {
+          CONFLICTS.push({
+            message: `This object was edited at path:${difference.path[0]}`,
+            category: 'schema',
+            object: { ...CURRENT_OBJECT },
+            additional_information: `Expected${difference.rhs}found${difference.lhs}`,
+          });
+        } else if (difference.kind === 'N') {
+          if (typeof CURRENT_OBJECT === 'undefined') {
+            ADDED.push({
+              message: 'This object was added.',
+              object: { ...NEW_OBJECT },
+            });
+          } else {
+            CONFLICTS.push({
+              message: 'This object was edited',
+              category: 'schema',
+              object: { ...NEW_OBJECT },
+            });
+          }
+        } else if (difference.kind === 'D' && difference.path[0] === 'list') {
+          CONFLICTS.push({
+            message: 'This object was edited, should be a list.',
+            category: 'schema',
+            object: { ...NEW_OBJECT },
+          });
+        } else if (typeof NEW_OBJECT !== 'undefined' && typeof NEW_OBJECT.index !== 'undefined' && NEW_OBJECT.index === true && difference.kind === 'A') {
+          CONFLICTS.push({
+            message: 'Tokenizer of this objects was edited.',
+            category: 'schema',
+            object: { ...CURRENT_OBJECT },
+          });
+        }
+      });
+    }
+  });
+  MISSING_PREDICATES.forEach(predicate => {
+    const DELETED_OBJECT = current_schema.schema.find(object => object.predicate === predicate);
+    CONFLICTS.push({
+      message: 'This object was deleted.',
+      object: { ...DELETED_OBJECT },
+    });
+  });
+
+  return [CONFLICTS, ADDED];
+}
+
+const raw_schema = () => {
+  let raw_predicates = '';
+  Object.entries(SCHEMA).forEach(([key, value]) => {
```

Comment:
Take SCHEMA as parameter

Reply:
eslint:  SCHEMA already declared in the upper scope

Reply:
lower case it, it will not conflict